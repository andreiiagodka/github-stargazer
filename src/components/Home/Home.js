import React, { Component } from "react";
import {
  Row,
  Col
} from "react-bootstrap";

import SearchForm from './SearchForm/SearchForm';
class Home extends Component {
  render() {
    return (
      <div>
        <Row>
          <Col>
            <SearchForm />
          </Col>
        </Row>
      </div>
    )
  }
}

export default Home