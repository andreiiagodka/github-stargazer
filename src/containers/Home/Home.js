import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { connect } from 'react-redux'
import * as actions from '../../store/actions/actions'

import Spinner from '../../components/UI/Spinner/Spinner'
import Header from '../../components/UI/Header/Header'
import Layout from '../../components/UI/Layout/Layout'
import ActionButton from '../../components/Home/ActionButton/ActionButton'
import Repositories from '../../components/Home/Repositories/Repositories'

class Home extends Component {
  componentDidMount() {
    this.props.fetchRepositories()
  }

  render() {
    let content = <Spinner />
    if (!this.props.loading) {
      const header = <Header title='Github Stargazer' actionButton={<ActionButton />} />
      const body = <Repositories repositories={this.props.repositories} />
      
      content = <Layout header={header} body={body} />
    }

    return content
  }
}

Home.propTypes = {
  repositories: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired,
  fetchRepositories: PropTypes.func.isRequired
}

const mapStateToProps = state => {
  return {
    repositories: state.fetchRepositories.repositories,
    loading: state.fetchRepositories.loading
  }
}

const mapDispatchToProps = dispatch => {
  return {
    fetchRepositories: () => dispatch(actions.fetchRepositories())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)