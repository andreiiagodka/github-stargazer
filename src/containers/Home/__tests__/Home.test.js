import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { MemoryRouter } from 'react-router'
import Home from '../Home'

describe('<Home /> container', () => {
  let store

  const mockStore = configureStore([])
  const fetchRepositories = {
    repositories: [],
    loading: false,
    fetchRepositories: jest.fn()
  }

  it('renders <Spinner /> container', () => {
    const fetchRepositoriesProps = {
      ...fetchRepositories,
      loading: true
    }
    store = mockStore({ fetchRepositories: fetchRepositoriesProps })
    store.dispatch = jest.fn()

    const tree = renderer.create(
      <Provider store={store} >
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders <Blank /> container', () => {
    store = mockStore({ fetchRepositories: fetchRepositories })
    store.dispatch = jest.fn()

    const tree = renderer.create(
      <Provider store={store} >
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders container correctly', () => {
    const fetchRepositoriesProps = {
      ...fetchRepositories,
      repositories: [
        {
          id: '1',
          full_name: 'Full name',
          stargazers_count: 1
        }
      ]
    }
    store = mockStore({ fetchRepositories: fetchRepositoriesProps })
    store.dispatch = jest.fn()

    const tree = renderer.create(
      <Provider store={store} >
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </Provider>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})