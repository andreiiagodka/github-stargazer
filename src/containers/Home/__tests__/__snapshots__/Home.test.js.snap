// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> container renders <Blank /> container 1`] = `
Array [
  <div
    className="card-header"
  >
    <div
      className="row"
    >
      <div
        className="col-md-2"
      />
      <div
        className="text-center col-md-8"
      >
        <h5>
          Github Stargazer
        </h5>
      </div>
      <div
        className="text-right col-md-2"
      >
        <a
          href="/new"
          onClick={[Function]}
        >
          <button
            className="btn btn-light"
            disabled={false}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus fa-w-14 "
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </a>
      </div>
    </div>
  </div>,
  <div
    className="card-body"
  >
    <div
      className="row"
    >
      <div
        className="text-center col-md-12"
      >
        <p
          className="lead"
        >
          No repositories added
        </p>
      </div>
    </div>
  </div>,
]
`;

exports[`<Home /> container renders <Spinner /> container 1`] = `
<div
  className="Loader"
>
  Loading ...
</div>
`;

exports[`<Home /> container renders container correctly 1`] = `
Array [
  <div
    className="card-header"
  >
    <div
      className="row"
    >
      <div
        className="col-md-2"
      />
      <div
        className="text-center col-md-8"
      >
        <h5>
          Github Stargazer
        </h5>
      </div>
      <div
        className="text-right col-md-2"
      >
        <a
          href="/new"
          onClick={[Function]}
        >
          <button
            className="btn btn-light"
            disabled={false}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus fa-w-14 "
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </a>
      </div>
    </div>
  </div>,
  <div
    className="card-body"
  >
    <div
      className="list-group"
      onKeyDown={[Function]}
    >
      <a
        href="/details/1"
        onClick={[Function]}
      >
        <button
          className="list-group-item list-group-item-action"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="row"
          >
            <div
              className="col-md-7"
            >
              Full name
            </div>
            <div
              className="text-right col-md-5"
            >
              1
            </div>
          </div>
        </button>
      </a>
    </div>
  </div>,
]
`;
